lib/synapse_pay_rest/http_client.rb

```diff
@@ -194,6 +159,69 @@ def delete(path)
 
     private
 
+    def request_params(method, url:, payload: nil, idempotency_key: nil)
+      request_headers = headers
+      if idempotency_key
+        request_headers.merge!({'X-SP-IDEMPOTENCY-KEY' => idempotency_key})
+      end
+
+      params = {
+        method: method,
+        url: url,
+        headers: request_headers
+      }
+
+      params[:payload] = payload.to_json if payload.present?
+      params
+    end
+
+    # Sends a requests
+    #
+    # @param method [Symbol]
+    # @param url [String]
+    # @param payload [Hash]
+    # @param idempotency_key [String]
+    #
+    # @return [String] Stringified version of the JSON response
```

Comment:
are you sure this is the return type? i thought this would return `RestClient::Response`